src/blob/index/bptree/core.rs

```diff
@@ -144,7 +144,9 @@ where
         // FIXME: check hash here?
         if !self.header.is_written() {
             let param = ValidationErrorKind::IndexIsWritten;
-            return Err(Error::validation(param, "Index is not written").into());
+            return Err(
+                Error::validation(param, "Index is incomplete (no 'is_written' flag)").into(),
```

Comment:
There were 2 places with the same message, but you only correct one of them. Please, correct the other one too